server/prepare.c

```diff
@@ -287,6 +288,20 @@ static void send_capture_info(
 
     struct xml_element element =
         start_element(file, "data", options->xml_header, false, true);
+    
+    // valerix begin
+    struct timespec ts;
+    struct tm tm;
+    char message[MAX_RESULT_LENGTH];
+    clock_gettime(CLOCK_REALTIME, &ts);
```

Comment:
Unfortunately, this timestamp is not captured at the right time: the enclosing function `send_capture_info()` is called during transmission of the captured stream to the client application, and this can result in different timestamps for different concurrent clients.

Instead, the ARM timestamp should be captured while processing `arm_capture()` (`data_server.c`), and then maybe can most simply be passed as a parameter to `send_data_header`.  It's safe to use static variables for this state.

Reply:
Thank you! I totally missed this. Fixed